js/index.js

```diff
@@ -1,37 +1,78 @@
+  product.querySelector('.subtotal span').innerHTML = subtotal
+
+  return subtotal
 }
 
-function calculateAll() {
-  // code in the following two lines is added just for testing purposes.
-  // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
-  // end of test
 
-  // ITERATION 2
-  //... your code goes here
+function calculateAll() {
+  const singleProduct = document.querySelector('.product')
+  updateSubtotal(singleProduct)
+ 
+  const productTotal = document.querySelectorAll('.product')
+  let total = 0
 
-  // ITERATION 3
-  //... your code goes here
+  productTotal.forEach(product => {
+      total += updateSubtotal(product)
+  })
+  let totalValue = document.querySelector('#total-value span').innerText = total.toFixed(2)
 }
 
 // ITERATION 4
+const removeBtn = document.querySelectorAll('.btn.btn-remove')
+removeBtn.forEach((button) => {
+  button.addEventListener('click', removeProduct)
+});
 
-function removeProduct(event) {
-  const target = event.currentTarget;
-  console.log('The target in remove is:', target);
-  //... your code goes here
+function removeProduct() {
+  const product = document.querySelector('.product')
+  product.parentNode.removeChild(product)
+
+  calculateAll()
 }
 
 // ITERATION 5
+createBtn = document.getElementById('create').addEventListener('click', createProduct)
 
-function createProduct() {
-  //... your code goes here
+function createProduct(product) {
+  const newProductTr = document.createElement('tr');
+  newProductTr.classList.add('product')
+  nameInput = document.querySelector('.create-product input[type=text]')
+  priceInput = document.querySelector('.create-product input[type=number]')
+
+  newProductName = document.createElement('td')
+  newProductName.innerHTML = `<span>${nameInput.value}</span>`
+  newProductName.classList.add('name')
+
+  newProductPrice = document.createElement('td')
+  newProductPrice.innerHTML = `$<span>${priceInput.value}</span>`
+  newProductPrice.classList.add('price')
+  
+  newProductQuantity = document.createElement('td')
+  newProductQuantity.innerHTML = '<input type="number" value="0" min="0" placeholder="Quantity" />'
+  newProductQuantity.classList.add('quantity')
+  
+  newProductSubtotal = document.createElement('td')
+  newProductSubtotal.innerHTML = `$<span>0</span>`
+  newProductSubtotal.classList.add('subtotal')
+  
+  newProductAction = document.createElement('td')
+  newProductAction.classList.add('action')
+  newProductAction.innerHTML = '<button class="btn btn-remove">Remove</button>'
+  newProductAction.addEventListener('click', removeProduct)
+
+  newProductTr.appendChild(newProductName)
+  newProductTr.appendChild(newProductPrice)
+
+  let appendedTr = tbody.appendChild(newProductTr)
+  appendedTr.appendChild(newProductQuantity)
+  appendedTr.appendChild(newProductSubtotal)
+  appendedTr.appendChild(newProductAction)
```

Comment:
Respect for this work! 💪 You could have also inserted the whole html-element in one string (to write it shorter)